controller/controller.go

```diff
@@ -305,6 +364,21 @@ func (c *controllerAPI) getRoute(ctx context.Context) (*router.Route, error) {
 	return route, err
 }
 
+func createEvent(dbExec func(string, ...interface{}) (sql.Result, error), appID, objectID string, objectType ct.EventType, data interface{}) error {
```

Comment:
Refactor to `func createEvent(dbExec func(string, ...interface{}) (sql.Result, error), e *ct.Event, data interface{}) error`